packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js

```diff
@@ -45,39 +47,69 @@ export default createConnector({
     const results = getResults(searchResults, this.context);
 
     this._allResults = this._allResults || [];
-    this._previousPage = this._previousPage || 0;
+    this._prevState = this._prevState || '';
```

Comment:
`this._prevState` is expected to be an object, right?

Reply:
Correct. It was previously serialized as a string and I forgot to change the initial value when I changed it's type.